language/src/modules/Language.rb

```diff
@@ -915,53 +916,20 @@ def Selection
 
     # Save state to target.
     def Save
-      loc = locale_to_install
-
-      @localed_conf = {} if @localed_conf.nil?
-
-      if Builtins.find(loc, "zh_HK") == 0
-        @localed_conf["LC_MESSAGES"] = "zh_TW"
-      elsif @localed_conf["LC_MESSAGES"] == "zh_TW"
-        # FIXME ugly hack: see bug #47711
-        @localed_conf.delete("LC_MESSAGES")
-      end
-
       SCR.Write(path(".sysconfig.language.INSTALLED_LANGUAGES"), @languages)
       SCR.Write(path(".sysconfig.language"), nil)
 
-      @localed_conf["LANG"] = loc
-      log.info("Locale: #{@localed_conf}")
-      locale_out1 = @localed_conf.map do | key, val |
-        "#{key}=#{val}"
-      end
-      log.info("Locale: #{locale_out1}")
-      locale_out=locale_out1.join(",")
-      log.info("Locale: #{locale_out}")
+      command = save_command_for(locale)
 
-      cmd = if Stage.initial
-        # do use --root option, running in chroot does not work
-        "/usr/bin/systemd-firstboot --root #{Installation.destdir.shellescape} --locale #{loc.shellescape}"
-      else
-        # this sets both the locale (see "man localectl")
-        "/usr/bin/localectl set-locale #{locale_out.shellescape}"
-      end
-      log.info "Making language setting persistent: #{cmd}"
-      result = if Stage.initial
-        WFM.Execute(path(".local.bash_output"), cmd)
-      else
-        SCR.Execute(path(".target.bash_output"), cmd)
-      end
-      if result["exit"] != 0
-        log.error "Language configuration not written. Failed to execute '#{cmd}'"
-        log.error "output: #{result.inspect}"
-        # TRANSLATORS: the "%s" is replaced by the executed command
-        Report.Error(_("Could not save the language setting, the command\n%s\nfailed.") % cmd)
-      else
-        log.info "output: #{result.inspect}"
-      end
+      log.info("Making language settings persistent: #{command}")
 
-      Builtins.y2milestone("Saved data for language: <%1>", loc)
+      Yast::Execute.locally!(command.split)
+      nil
+    rescue Cheetah::ExecutionFailed => e
+      log.error "Language configuration not written.\n#{e.inspect}"
 
+      # TRANSLATORS: the "%s" is replaced by the executed command
+      Report.Error(_("Could not save the language setting, the command\n%s\nfailed.") % command)
```

Comment:
please try to write also stderr to give user hint why it failed. Not just what.

Reply:
I think that the "stderr" do not give to the user (I mean, normal user) too much information. User who find the "stderr" useful probably will look for it in the logs, isn't it?

That said, I did not touch this intentionally because the "text freeze". So, for now, I prefer to keep it as it is/was.

Reply:
ok, text freeze is good point. There are users which understand stderr, but do not look immediatelly to logs. And it also really helps in openqa as you see it in screenshot.